sdk/keyvault/keyvault-keys/review/keyvault-keys.review.api.md

```diff
@@ -0,0 +1,295 @@
+    result: Uint8Array;
+}
+
+// @public
+export interface GetKeyOptions {
+    requestOptions?: msRest.RequestOptionsBase;
+    version?: string;
+}
+
+// @public
+export interface GetKeysOptions {
+    requestOptions?: msRest.RequestOptionsBase;
+}
+
+// @public
+export interface ImportKeyOptions {
+    enabled?: boolean;
+    expires?: Date;
+    hsm?: boolean;
+    notBefore?: Date;
+    requestOptions?: msRest.RequestOptionsBase;
+    tags?: {
+        [propertyName: string]: string;
+    };
+}
+
+// @public
+export interface JsonWebKey {
+    crv?: JsonWebKeyCurveName;
+    d?: Uint8Array;
+    dp?: Uint8Array;
+    dq?: Uint8Array;
+    e?: Uint8Array;
+    k?: Uint8Array;
+    // (undocumented)
+    keyOps?: string[];
+    kid?: string;
+    kty?: JsonWebKeyType;
+    n?: Uint8Array;
+    p?: Uint8Array;
+    q?: Uint8Array;
+    qi?: Uint8Array;
+    t?: Uint8Array;
+    x?: Uint8Array;
+    y?: Uint8Array;
+}
+
+// @public
+export type JsonWebKeyCurveName = "P-256" | "P-384" | "P-521" | "P-256K";
+
+// @public
+export type JsonWebKeyEncryptionAlgorithm = "RSA-OAEP" | "RSA-OAEP-256" | "RSA1_5";
+
+// @public
+export type JsonWebKeyOperation = "encrypt" | "decrypt" | "sign" | "verify" | "wrapKey" | "unwrapKey";
+
+// @public
+export type JsonWebKeyType = "EC" | "EC-HSM" | "RSA" | "RSA-HSM" | "oct";
+
+// @public
+export interface Key {
+    keyMaterial?: JsonWebKey;
+    properties: KeyProperties;
+}
+
+// @public
+export interface KeyProperties extends ParsedKeyVaultEntityIdentifier {
+    readonly created?: Date;
+    enabled?: boolean;
+    expires?: Date;
+    id?: string;
+    notBefore?: Date;
+    readonly recoveryLevel?: DeletionRecoveryLevel;
+    tags?: {
+        [propertyName: string]: string;
+    };
+    readonly updated?: Date;
+}
+
+// @public
+export class KeysClient {
+    constructor(url: string, credential: TokenCredential, pipelineOrOptions?: ServiceClientOptions | NewPipelineOptions);
+    backupKey(name: string, options?: RequestOptions): Promise<Uint8Array | undefined>;
+    createEcKey(name: string, options?: CreateEcKeyOptions): Promise<Key>;
+    createKey(name: string, keyType: JsonWebKeyType, options?: CreateKeyOptions): Promise<Key>;
+    createRsaKey(name: string, options?: CreateRsaKeyOptions): Promise<Key>;
+    protected readonly credential: TokenCredential;
+    deleteKey(name: string, options?: RequestOptions): Promise<DeletedKey>;
+    static getDefaultPipeline(credential: TokenCredential, pipelineOptions?: NewPipelineOptions): ServiceClientOptions;
+    getDeletedKey(name: string, options?: RequestOptions): Promise<DeletedKey>;
+    getKey(name: string, options?: GetKeyOptions): Promise<Key>;
+    importKey(name: string, key: JsonWebKey, options?: ImportKeyOptions): Promise<Key>;
+    listDeletedKeys(options?: GetKeysOptions): PagedAsyncIterableIterator<KeyProperties, KeyProperties[]>;
+    listKeys(options?: GetKeysOptions): PagedAsyncIterableIterator<KeyProperties, KeyProperties[]>;
```

Comment:
There is a discussion across languages and key vault libraries as to what the naming of this method.  Since it returns an iterable of KeyProperties rather than Keys should it be named listKeysProperties?listAllKeysProperties?